contracts-periphery/src/UniswapWithdrawHook.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+pragma abicoder v2;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "src/interfaces/ISwapRouter.sol";
+
+contract UniswapWithdrawHook is Ownable {
+  using SafeERC20 for IERC20;
+
+  ISwapRouter internal immutable swapRouter;
+  address internal WETH9;
+
+  uint256 public feeBips;
+  address public feeCollector;
+  address payable public feeReceiver;
+
+  constructor(ISwapRouter _swapRouter, address _WETH9, uint256 _feeBips, address payable _feeReceiver) {
+  swapRouter = _swapRouter;
+  WETH9 = _WETH9;
+  feeBips = _feeBips;
+  feeReceiver = _feeReceiver;
+  }
+
+  function tokensWithdrawn(
+    uint256 _amount,
+    address _stealthAddr,
+    address _acceptor,
+    address _tokenAddr,
+    address _sponsor,
+    uint256 _sponsorFee,
+    bytes memory _data
+  ) external{
+
+    (bool _singleTrade, , ) = abi.decode(_data, (bool, address, ISwapRouter.ExactInputParams[]));
+    if(_singleTrade) {
+      _swapPartForEth(_amount, _stealthAddr, _acceptor, _tokenAddr, _sponsor, _sponsorFee, _data);
+    } else {
+      _swapPartForTokenPartForEth(_amount, _stealthAddr, _acceptor, _tokenAddr, _sponsor, _sponsorFee, _data);
+    }
+
+  }
+
+  function _swapPartForEth(
```

Comment:
IIUC, this method swaps some of the tokens for ETH, then sends the rest of the tokens to the same recipient? If so sounds good and makes sense

Reply:
Yes